### PR TITLE
mgr/dashboard: added cephfs mount details

### PR DESCRIPTION
**Pull request by ivoalmeida**
Fixes: https://tracker.ceph.com/issues/64405

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

[Screencast from 2024-02-13 12-03-51.webm](https://github.com/ceph/ceph/assets/1071297/ded5ca0e-bdd0-4d3a-95eb-e14edda1163d)

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-list/cephfs-list.component.ts [hunk 1]
```diff
@@ -95,6 +98,13 @@ export class CephfsListComponent extends ListWithDetails implements OnInit {
         click: () => this.removeVolumeModal(),
         name: this.actionLabels.REMOVE,
         disable: this.getDisableDesc.bind(this)
+      },
+      {
+        name: this.actionLabels.ATTACH,
+        permission: 'read',
+        icon: Icons.bars,
+        disable: () => !this.selection?.hasSelection,
+        click: () => this.showAttachInfo()
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-list/cephfs-list.component.ts:107]: this should have precedence before `Remove` and after `Edit`. `Remove` should be the final action in the list.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-mount-details/cephfs-mount-details.component.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      <h5 class="fw-bold"
+          i18n>
+        Using Mount command
+      </h5>
+      <prep [ngClass]="['hstack', 'gap-3','text-wrap', 'p-4', 'mb-4', 'code-block']">
+        <code id="mount">
+          <span>sudo &lt;client_user&gt;&#64;{{ mountData?.fsId }}.{{ mountData?.fsName }}={{ mountData?.rootPath }} &lt;mount/point&gt;</span>
+        </code>
+        <cd-copy-2-clipboard-button source="mount"></cd-copy-2-clipboard-button>
+      </prep>
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-mount-details/cephfs-mount-details.component.html:16]: we have a [code-block](https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/frontend/src/app/shared/components/code-block/code-block.component.html). can we reuse that here?

Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-mount-details/cephfs-mount-details.component.html:16]: you can improve it if you want

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-list/cephfs-subvolume-list.component.ts [hunk 3]
```diff
@@ -154,6 +156,13 @@ export class CephfsSubvolumeListComponent extends CdForm implements OnInit, OnCh
         permission: 'delete',
         icon: Icons.destroy,
         click: () => this.removeSubVolumeModal()
+      },
+      {
+        name: this.actionLabels.ATTACH,
+        permission: 'read',
+        icon: Icons.bars,
+        disable: () => !this.selection?.hasSelection,
+        click: () => this.showAttachInfo()
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-list/cephfs-subvolume-list.component.ts:165]: same here, `Remove` should be the last action!

**Review by nizamial09 — APPROVED** [review]
Looks mostly ok to me, thank you @ivoalmeida

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-list/cephfs-list.component.ts (hunk 1 above)
Comment (ivoalmeida) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-list/cephfs-list.component.ts:103]: done

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-mount-details/cephfs-mount-details.component.html (hunk 2 above)
Comment (ivoalmeida) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-mount-details/cephfs-mount-details.component.html]: done

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-list/cephfs-subvolume-list.component.ts (hunk 3 above)
Comment (ivoalmeida) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-list/cephfs-subvolume-list.component.ts:160]: done

**Review by cloudbehl — APPROVED** [review]
Tested it, working as expected.

Some tests are failing.

**Comment by cloudbehl** [comment]
We will need to have way to download the `ceph.conf` and cephx user config file. Which is required as a pre req for this to work. @ivoalmeida can you create tracker for this?

**Comment by ivoalmeida** [comment]
> We will need to have way to download the `ceph.conf` and cephx user config file. Which is required as a pre req for this to work. @ivoalmeida can you create tracker for this?

Sure! 

https://tracker.ceph.com/issues/64422

**Comment by ivoalmeida** [comment]
> Tested it, working as expected.
> 
> Just the cephfuse is not changing the dir path based on the subvolume selected. In the below image the fuse should be pointing to the subvolume path.
> 
> image
> 
> Some tests are failing.

Fixed the path for cephfuse and the test failing.

image

**Comment by nizamial09** [comment]
@ivoalmeida you might not be happy but
```
[lint:html    ] 
[lint:html    ] [htmllint] found 0 errors out of 213 files
[lint:html    ] All files pass linting
[lint:prettier] src/app/ceph/cephfs/cephfs-mount-details/cephfs-mount-details.component.ts
[lint:prettier] src/app/ceph/cephfs/cephfs-snapshotschedule-list/cephfs-snapshotschedule-list.component.ts
[lint:eslint  ] 
[lint:eslint  ] Linting "ceph-dashboard"...
[lint:eslint  ] 
[lint:eslint  ] All files pass linting.
[lint:eslint  ] 

Try running 'npm run fix' to fix some linting errors. Some errors might need a manual fix.
```

**Pull request merged**